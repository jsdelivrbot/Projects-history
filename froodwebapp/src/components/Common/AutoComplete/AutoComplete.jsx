import React from 'react';
import PropTypes from 'prop-types';
import ReactAutoComplete from 'react-autocomplete';
import {
  autoComplete,
  autoCompleteMenuNoItems,
  autoCompleteMenu,
  autoCompleteInput
} from './AutoComplete.scss';

const AutoComplete = ({
  // trigger
  loadingAutoComplete,
  // data
  items,
  value,
  // props
  getItemValue,
  inputRef,
  inputPlaceholder,
  inputStyle = {},
  renderItem,
  alignRight = false,
  disabled = false,
  // handlers
  onChange,
  onSelect,
  onBlur,
  onFocus
}) => (
  <ReactAutoComplete
    ref={ inputRef }
    wrapperProps={ {
      className: autoComplete
    } }
    autoHighlight
    items={ items }
    getItemValue={ getItemValue }
    value={ value }
    onChange={ onChange }
    onSelect={ onSelect }
    renderItem={ renderItem }
    renderMenu={ (renderItems, renderValue) => (
      <div>
        { loadingAutoComplete && <div className={ autoCompleteMenuNoItems }>Loading...</div> }
        {
          !loadingAutoComplete
          && renderItems.length === 0
          && renderValue.length > 2
          && <div className={ autoCompleteMenuNoItems }>No matches for {value}</div>
        }
        {
          !loadingAutoComplete
          && renderItems.length !== 0
          && renderValue.length > 2
          &&
            <div
              className={ autoCompleteMenu }
              style={ { right: alignRight ? 0 : 'auto' } }
            >
              { renderItems }
            </div>
        }
      </div>
    ) }
    inputProps={ {
      lang: 'en',
      placeholder: inputPlaceholder,
      onBlur,
      onFocus,
      disabled,
      className: autoCompleteInput,
      style: {
        backgroundColor: disabled ? '#f7f7f7' : 'white',
        color: disabled ? 'rgba(0, 0, 0, 0.25)' : 'inherit',
        cursor: disabled && 'not-allowed',
        ...inputStyle
      }
    } }
  />
);

AutoComplete.propTypes = {
  // triggers
  loadingAutoComplete: PropTypes.bool.isRequired,
  disabled: PropTypes.bool,
  // data
  items: PropTypes.array.isRequired,
  value: PropTypes.string,
  // props
  inputRef: PropTypes.func,
  getItemValue: PropTypes.func.isRequired,
  renderItem: PropTypes.func.isRequired,
  inputStyle: PropTypes.object,
  inputPlaceholder: PropTypes.string,
  alignRight: PropTypes.bool,
  // handlers
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func
};

export default AutoComplete;
