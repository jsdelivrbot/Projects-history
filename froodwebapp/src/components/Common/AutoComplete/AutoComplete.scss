.autoComplete {
  display: block !important;
  height: 100%;
}

.autoCompleteMenu {
  z-index: 50;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.3);
  background: white;
  margin: 2px 2px;
  font-size: 0.8rem;
  position: absolute;
  display: block;
  overflow: auto;
  height: auto;
  max-height: 40vh;
  width: auto;
  white-space: nowrap;
  padding: 0.2rem;
}

.autoCompleteMenuNoItems {
  @extend .autoCompleteMenu;
  height: auto;
  width: auto;
  white-space: nowrap;
  overflow: hidden;
}

.autoCompleteInput {
  width: 100%;
  height: 28px;
  padding: 4px 7px;
  border-radius: 4px;
  font-size: 12px;
  line-height: 1.5;
  border-left: 1px solid lightgray;
  border-right: 1px solid lightgray;
  border-top: 1px solid lightgray;
  border-bottom: 1px solid lightgray;
  outline: none;
}