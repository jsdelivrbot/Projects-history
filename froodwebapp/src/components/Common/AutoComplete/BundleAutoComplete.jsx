import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AutoComplete } from 'components';
import { connect } from 'react-redux';
import {
  bundleSearchRequest
} from 'redux-base/actions';
import {
  renderAutocompleteItem
} from 'utils';

const getItemValue = item => item.id.toString();

const renderBundleItem = renderAutocompleteItem(['id', 'name', 'skuCode', 'locationName'], 'Sku');

const mapStateToProps = state => ({
  loadingAutoComplete: state.autocomplete.loadingAutoComplete,
  bundles: state.autocomplete.bundles,
});

const mapDispatchToProps = {
  bundleSearchRequest
};

export class BundleAutoComplete extends Component {
  onChange = (e, value) => {
    this.props.onChange(value);
    this.props.bundleSearchRequest({
      payload: value
    });
  }

  onSelect = (id) => {
    const { bundles } = this.props;
    const { name } = bundles.find(item => item.id === id);
    this.props.onSelect(id, name);
  }

  onFocus = () => {
    if (this.props.value) {
      this.props.bundleSearchRequest({
        payload: this.props.value
      });
    }
  }

  render() {
    const {
      value,
      loadingAutoComplete,
      bundles,
      alignRight = false,
      disabled = false
    } = this.props;

    return (
      <AutoComplete
        items={ bundles }
        getItemValue={ getItemValue }
        value={ value }
        onChange={ this.onChange }
        onSelect={ this.onSelect }
        onFocus={ this.onFocus }
        renderItem={ renderBundleItem }
        loadingAutoComplete={ loadingAutoComplete }
        inputPlaceholder="Bundle"
        alignRight={ alignRight }
        disabled={ disabled }
      />
    );
  }
}

BundleAutoComplete.propTypes = {
  // trigger
  loadingAutoComplete: PropTypes.bool.isRequired,
  // props
  value: PropTypes.string,
  bundles: PropTypes.array.isRequired,
  alignRight: PropTypes.bool,
  disabled: PropTypes.bool,
  // redux-base
  bundleSearchRequest: PropTypes.func.isRequired,
  // handlers
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(BundleAutoComplete);
