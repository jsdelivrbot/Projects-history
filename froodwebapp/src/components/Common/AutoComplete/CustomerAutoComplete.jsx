import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AutoComplete } from 'components';
import { connect } from 'react-redux';
import {
  customerSearchRequest
} from 'redux-base/actions';
import {
  renderAutocompleteItem
} from 'utils';

const getItemValue = item => item.id.toString();

const renderCustomerItem = renderAutocompleteItem(['id', 'name', 'email'], 'Customer');

const mapStateToProps = state => ({
  loadingAutoComplete: state.autocomplete.loadingAutoComplete,
  customers: state.autocomplete.customers
});

const mapDispatchToProps = {
  customerSearchRequest
};

export class CustomerAutoComplete extends Component {
  onChange = (e, value) => {
    this.props.onChange(value);
    this.props.customerSearchRequest({
      payload: value
    });
  }

  onSelect = (id) => {
    const { name } = this.props.customers.find(item => item.id === Number(id));
    this.props.onSelect(id, name);
  }

  onFocus = () => {
    if (this.props.value) {
      this.props.customerSearchRequest({
        payload: this.props.value
      });
    }
  }

  render() {
    const {
      value,
      loadingAutoComplete,
      customers,
      alignRight = false,
      disabled = false
    } = this.props;

    return (
      <AutoComplete
        items={ customers }
        getItemValue={ getItemValue }
        value={ value }
        onChange={ this.onChange }
        onSelect={ this.onSelect }
        onFocus={ this.onFocus }
        renderItem={ renderCustomerItem }
        loadingAutoComplete={ loadingAutoComplete }
        inputPlaceholder="Search Customer"
        alignRight={ alignRight }
        disabled={ disabled }
      />
    );
  }
}

CustomerAutoComplete.propTypes = {
  // trigger
  loadingAutoComplete: PropTypes.bool.isRequired,
  // props
  value: PropTypes.string,
  customers: PropTypes.array.isRequired,
  alignRight: PropTypes.bool,
  disabled: PropTypes.bool,
  // redux-base
  customerSearchRequest: PropTypes.func.isRequired,
  // handlers
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(CustomerAutoComplete);
