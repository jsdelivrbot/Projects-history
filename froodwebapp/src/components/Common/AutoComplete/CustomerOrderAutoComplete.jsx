import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AutoComplete } from 'components';
import { connect } from 'react-redux';
import {
  customerOrderSearchRequest
} from 'redux-base/actions';
import {
  renderAutocompleteItem
} from 'utils';

const getItemValue = item => item.id.toString();

const renderCustomerOrderItem = renderAutocompleteItem(['id', 'orderDate', 'status'], 'Order');

const mapStateToProps = state => ({
  loadingAutoComplete: state.autocomplete.loadingAutoComplete,
  customerOrders: state.autocomplete.customerOrders
});

const mapDispatchToProps = {
  customerOrderSearchRequest
};

export class CustomerOrderAutoComplete extends Component {
  onChange = (e, value) => {
    this.props.onChange(value);
    this.props.customerOrderSearchRequest({
      id: this.props.customerId,
      payload: value
    });
  }

  onSelect = (id) => {
    const { name } = this.props.customerOrders.find(item => item.id === Number(id));
    this.props.onSelect(id, name);
  }

  onFocus = () => {
    if (this.props.value) {
      this.props.customerOrderSearchRequest({
        id: this.props.customerId,
        payload: this.props.value
      });
    }
  }

  render() {
    const {
      value,
      loadingAutoComplete,
      customerOrders,
      alignRight = false,
      disabled = false
    } = this.props;

    return (
      <AutoComplete
        items={ customerOrders }
        getItemValue={ getItemValue }
        value={ value }
        onChange={ this.onChange }
        onSelect={ this.onSelect }
        onFocus={ this.onFocus }
        renderItem={ renderCustomerOrderItem }
        loadingAutoComplete={ loadingAutoComplete }
        inputPlaceholder="Search Orders"
        alignRight={ alignRight }
        disabled={ disabled }
      />
    );
  }
}

CustomerOrderAutoComplete.propTypes = {
  // trigger
  loadingAutoComplete: PropTypes.bool.isRequired,
  // props
  customerId: PropTypes.number.isRequired,
  value: PropTypes.string,
  customerOrders: PropTypes.array.isRequired,
  alignRight: PropTypes.bool,
  disabled: PropTypes.bool,
  // redux-base
  customerOrderSearchRequest: PropTypes.func.isRequired,
  // handlers
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(CustomerOrderAutoComplete);
