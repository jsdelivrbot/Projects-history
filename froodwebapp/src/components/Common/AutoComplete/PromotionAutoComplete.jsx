import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AutoComplete } from 'components';
import { connect } from 'react-redux';
import {
  promotionSearchRequest
} from 'redux-base/actions';
import {
  renderAutocompleteItem
} from 'utils';

const getItemValue = item => item.id.toString();

const renderPromotionItem = renderAutocompleteItem(['id', 'code', 'name'], 'Promotion');

const mapStateToProps = state => ({
  loadingAutoComplete: state.autocomplete.loadingAutoComplete,
  promotions: state.autocomplete.promotions
});

const mapDispatchToProps = {
  promotionSearchRequest
};

class PromotionAutoComplete extends Component {
  onChange = (e, value) => {
    this.props.onChange(value);
    this.props.promotionSearchRequest({
      payload: value
    });
  }

  onSelect = (id) => {
    const { code } = this.props.promotions.find(item => item.id === Number(id));
    this.props.onSelect(id, code);
  }

  onFocus = () => {
    if (this.props.value) {
      this.props.promotionSearchRequest({
        payload: this.props.value
      });
    }
  }

  render() {
    const {
      value,
      loadingAutoComplete,
      promotions,
      alignRight = false,
      disabled = false
    } = this.props;

    return (
      <AutoComplete
        items={ promotions }
        getItemValue={ getItemValue }
        value={ value }
        onChange={ this.onChange }
        onSelect={ this.onSelect }
        onFocus={ this.onFocus }
        renderItem={ renderPromotionItem }
        loadingAutoComplete={ loadingAutoComplete }
        inputPlaceholder="Type to search promotion"
        alignRight={ alignRight }
        disabled={ disabled }
      />
    );
  }
}

PromotionAutoComplete.propTypes = {
  // trigger
  loadingAutoComplete: PropTypes.bool.isRequired,
  // props
  value: PropTypes.string,
  promotions: PropTypes.array.isRequired,
  alignRight: PropTypes.bool,
  disabled: PropTypes.bool,
  // redux-base
  promotionSearchRequest: PropTypes.func.isRequired,
  // handlers
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(PromotionAutoComplete);
