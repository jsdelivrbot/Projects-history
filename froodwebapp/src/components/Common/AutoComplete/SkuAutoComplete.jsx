import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AutoComplete } from 'components';
import { connect } from 'react-redux';
import {
  skuSearchRequest,
  skuSearchByVendorRequest
} from 'redux-base/actions';
import { renderAutocompleteItem } from 'utils';

const getItemValue = item => item.id.toString();

const renderSkusItem = renderAutocompleteItem(['id', 'name', 'skuCode'], 'Sku/Variants');

const mapStateToProps = state => ({
  loadingAutoComplete: state.autocomplete.loadingAutoComplete,
  skus: state.autocomplete.skus
});

const mapDispatchToProps = {
  skuSearchRequest,
  skuSearchByVendorRequest
};

export class SkuAutoComplete extends Component {
  onChange = (e, value) => {
    if (this.props.vendorId) {
      this.props.skuSearchByVendorRequest({
        vendorId: this.props.vendorId,
        payload: value
      });
    } else {
      this.props.skuSearchRequest({
        payload: value
      });
    }
    this.props.onChange(value);
  }

  onSelect = (id) => {
    const { skus } = this.props;
    const { name } = skus.find(item => item.id === id);
    this.props.onSelect(id, name);
  }

  onFocus = () => {
    const {
      value,
      vendorId
    } = this.props;

    if (value && vendorId) {
      this.props.skuSearchByVendorRequest({
        vendorId,
        payload: value
      });
    } else if (value) {
      this.props.skuSearchRequest({
        payload: value
      });
    }
  }

  render() {
    const {
      value,
      loadingAutoComplete,
      skus,
      alignRight = false,
      disabled = false,
      inputStyle
    } = this.props;

    return (
      <AutoComplete
        items={ skus }
        getItemValue={ getItemValue }
        value={ value }
        onChange={ this.onChange }
        onSelect={ this.onSelect }
        onFocus={ this.onFocus }
        renderItem={ renderSkusItem }
        loadingAutoComplete={ loadingAutoComplete }
        inputPlaceholder="Start typing SKU or name..."
        alignRight={ alignRight }
        disabled={ disabled }
        inputStyle={ inputStyle }
      />
    );
  }
}

SkuAutoComplete.propTypes = {
  // trigger
  loadingAutoComplete: PropTypes.bool.isRequired,
  // props
  value: PropTypes.string,
  vendorId: PropTypes.number,
  skus: PropTypes.array.isRequired,
  alignRight: PropTypes.bool,
  disabled: PropTypes.bool,
  inputStyle: PropTypes.object,
  // redux-base
  skuSearchRequest: PropTypes.func.isRequired,
  skuSearchByVendorRequest: PropTypes.func.isRequired,
  // handlers
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(SkuAutoComplete);
