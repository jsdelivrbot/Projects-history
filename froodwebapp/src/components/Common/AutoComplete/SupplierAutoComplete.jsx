import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AutoComplete } from 'components';
import { connect } from 'react-redux';
import {
  supplierSearchRequest
} from 'redux-base/actions';
import {
  renderAutocompleteItem
} from 'utils';

const getItemValue = item => item.id.toString();

const renderSkusItem = renderAutocompleteItem(['id', 'name', 'country'], 'Supplier');

const mapStateToProps = state => ({
  loadingAutoComplete: state.autocomplete.loadingAutoComplete,
  suppliers: state.autocomplete.suppliers
});

const mapDispatchToProps = {
  supplierSearchRequest
};

export class SupplierAutoComplete extends Component {
  onChange = (e, value) => {
    this.props.onChange(value);
    this.props.supplierSearchRequest({
      payload: value
    });
  }

  onSelect = (id) => {
    const { name } = this.props.suppliers.find(item => item.id === Number(id));
    this.props.onSelect(id, name);
  }

  onFocus = () => {
    if (this.props.value) {
      this.props.supplierSearchRequest({
        payload: this.props.value
      });
    }
  }

  render() {
    const {
      value,
      loadingAutoComplete,
      suppliers,
      alignRight = false,
      disabled = false
    } = this.props;

    return (
      <AutoComplete
        items={ suppliers }
        getItemValue={ getItemValue }
        value={ value }
        onChange={ this.onChange }
        onSelect={ this.onSelect }
        onFocus={ this.onFocus }
        renderItem={ renderSkusItem }
        loadingAutoComplete={ loadingAutoComplete }
        inputPlaceholder="Search Supplier"
        alignRight={ alignRight }
        disabled={ disabled }
      />
    );
  }
}

SupplierAutoComplete.propTypes = {
  // trigger
  loadingAutoComplete: PropTypes.bool.isRequired,
  // props
  value: PropTypes.string,
  suppliers: PropTypes.array.isRequired,
  alignRight: PropTypes.bool,
  disabled: PropTypes.bool,
  // redux-base
  supplierSearchRequest: PropTypes.func.isRequired,
  // handlers
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(SupplierAutoComplete);
