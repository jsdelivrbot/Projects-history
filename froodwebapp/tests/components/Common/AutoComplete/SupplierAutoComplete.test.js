import React from 'react';
import { SupplierAutoComplete } from 'components/Common/AutoComplete/SupplierAutoComplete';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, it, expect } = global;

describe('SupplierAutoComplete', () => {
  it('captures snapshot of SupplierAutoComplete', () => {
    const props = {
      loadingAutoComplete: false,
      value: 'value',
      suppliers: [],
      alignRight: false,
      disabled: false,
      supplierSearchRequest: jest.fn(),
      onChange: jest.fn(),
      onSelect: jest.fn()
    };

    const snappedComponent = shallow(<SupplierAutoComplete { ...props } />);

    expect(toJson(snappedComponent)).toMatchSnapshot();
  });
});
