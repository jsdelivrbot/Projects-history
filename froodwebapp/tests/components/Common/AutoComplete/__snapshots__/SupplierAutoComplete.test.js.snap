// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierAutoComplete captures snapshot of SupplierAutoComplete 1`] = `
<AutoComplete
  alignRight={false}
  disabled={false}
  getItemValue={[Function]}
  inputPlaceholder="Search Supplier"
  items={Array []}
  loadingAutoComplete={false}
  onChange={[Function]}
  onFocus={[Function]}
  onSelect={[Function]}
  renderItem={[Function]}
  value="value"
/>
`;
